pgrx-tests/src/tests/struct_type_tests.rs

```diff
@@ -16,13 +16,31 @@ use pgrx::AllocatedByRust;
 
 use crate::get_named_capture;
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 #[repr(C)]
-struct Complex {
+pub struct Complex {
     x: f64,
     y: f64,
 }
 
+impl Eq for Complex {}
+impl PartialEq for Complex {
+    fn eq(&self, other: &Self) -> bool {
+        self.x == other.x && self.y == other.y
+    }
+}
+
+impl Complex {
+    pub fn random() -> PgBox<Complex> {
+        unsafe {
+            let mut c = PgBox::<Complex>::alloc0();
+            c.x = rand::random();
+            c.y = rand::random();
+            c.into_pg_boxed()
+        }
+    }
+}
```

Comment:
Mmm I don't know if I like this.

Reply:
It's just generating some random test data.  If you got a better idea, I'm all ears.  Not sure how much any of has has to "like" it?

Reply:
`Box<Complex>` should be fine if it's just for testing? We can just leak allocations, it's not a big deal.

Reply:
`Box<Complex>` isn't a thing that we can FromDatum/IntoDatum.  That's why this does `PgBox<Complex>`.  I thought your complaint was about the `rand` usage!

Reply:
Oh, you're right.